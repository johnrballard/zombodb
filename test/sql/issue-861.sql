select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* and body:*') order by 1, 2, 3;
select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* and body:java') order by 1, 2, 3;
select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:42 and body:*') order by 1, 2, 3;
select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* or body:*') order by 1, 2, 3;
select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* or body:java') order by 1, 2, 3;
select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:42 or body:*') order by 1, 2, 3;