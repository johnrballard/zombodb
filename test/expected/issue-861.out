select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* and body:*') order by 1, 2, 3;
 field_name | array_index | term |      type      | position | start_offset | end_offset | query_clause 
------------+-------------+------+----------------+----------+--------------+------------+--------------
 body       |           0 | java | <ALPHANUM>     |        1 |            0 |          4 | body:*
 id         |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | id:*
(2 rows)

select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* and body:java') order by 1, 2, 3;
 field_name | array_index | term |      type      | position | start_offset | end_offset | query_clause 
------------+-------------+------+----------------+----------+--------------+------------+--------------
 body       |           0 | java | <ALPHANUM>     |        1 |            0 |          4 | body:"java"
 id         |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | id:*
(2 rows)

select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:42 and body:*') order by 1, 2, 3;
 field_name | array_index | term |      type      | position | start_offset | end_offset | query_clause 
------------+-------------+------+----------------+----------+--------------+------------+--------------
 body       |           0 | java | <ALPHANUM>     |        1 |            0 |          4 | body:*
 id         |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | id:"42"
(2 rows)

select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* or body:*') order by 1, 2, 3;
 field_name | array_index | term |      type      | position | start_offset | end_offset | query_clause 
------------+-------------+------+----------------+----------+--------------+------------+--------------
 body       |           0 | java | <ALPHANUM>     |        1 |            0 |          4 | body:*
 id         |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | id:*
(2 rows)

select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:* or body:java') order by 1, 2, 3;
 field_name | array_index | term |      type      | position | start_offset | end_offset | query_clause 
------------+-------------+------+----------------+----------+--------------+------------+--------------
 body       |           0 | java | <ALPHANUM>     |        1 |            0 |          4 | body:"java"
 id         |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | id:*
(2 rows)

select * from zdb.highlight_document('so_posts', '{"body": "java", "id":42}'::json, 'id:42 or body:*') order by 1, 2, 3;
 field_name | array_index | term |      type      | position | start_offset | end_offset | query_clause 
------------+-------------+------+----------------+----------+--------------+------------+--------------
 body       |           0 | java | <ALPHANUM>     |        1 |            0 |          4 | body:*
 id         |           0 | 42   | <UNSIGNED_INT> |        1 |            0 |          0 | id:"42"
(2 rows)

